### PR TITLE
python3Packages.sip: 6.8.3 -> 6.8.6

### PR DESCRIPTION
**Pull request by fabianhjr**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabianhjr** [comment]
cc @nrdxp

File: pkgs/development/python-modules/sip/default.nix [hunk 1]
```diff
@@ -15,19 +16,20 @@
...
+  version = "6.8.6";
   pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-iIVHsBi7JMNq3tUZ6T0+UT1MaqC6VbfMGv+9Rc8Qdiw=";
+    hash = "sha256-f8lZ5I5uxdWvi9Am9p9eJNCLPLirs0IXb1q4AwzAfXo=";
   };
 
   nativeBuildInputs = [ setuptools ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/sip/default.nix:27]:
```suggestion
  build-system = [ setuptools ];
```

File: pkgs/development/python-modules/sip/default.nix [hunk 2]
```diff
@@ -15,19 +16,20 @@
...
 
   propagatedBuildInputs = [
```
Comment (dotlambda) [inline pkgs/development/python-modules/sip/default.nix:29]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/sip/default.nix [hunk 3]
```diff
@@ -15,19 +16,20 @@
...
     packaging
     setuptools
+    setuptools-scm
```
Comment (dotlambda) [inline pkgs/development/python-modules/sip/default.nix:32]: Why is this propagated?

File: pkgs/development/python-modules/sip/default.nix [hunk 4]
```diff
@@ -15,19 +16,20 @@
...
   ] ++ lib.optionals (pythonOlder "3.11") [ tomli ];
 
   # There aren't tests
```
Comment (dotlambda) [inline pkgs/development/python-modules/sip/default.nix:35]: Not true.

File: pkgs/development/python-modules/sip/default.nix (hunk 3 above)
Comment (fabianhjr) [reply pkgs/development/python-modules/sip/default.nix:32]: Followed the same setuptools pattern assuming it had to be propagated along setuptools

File: pkgs/development/python-modules/sip/default.nix (hunk 2 above)
Comment (fabianhjr) [reply pkgs/development/python-modules/sip/default.nix:29]: Haven't read up on this changes. Could you ref an overview/doc on these changes for *inputs to dependencies/build-system?

File: pkgs/development/python-modules/sip/default.nix (hunk 4 above)
Comment (fabianhjr) [reply pkgs/development/python-modules/sip/default.nix:35]: dupe of https://github.com/NixOS/nixpkgs/pull/347606#discussion_r1794401301

File: pkgs/development/python-modules/sip/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/sip/default.nix:29]: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/sip/default.nix (hunk 4 above)
Comment (dotlambda) [reply pkgs/development/python-modules/sip/default.nix:35]: How is that comment related to the existence of tests?

File: pkgs/development/python-modules/sip/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/python-modules/sip/default.nix:32]: It's not in `dependencies`: https://github.com/Python-SIP/sip/blob/main/pyproject.toml#L15

File: pkgs/development/python-modules/sip/default.nix (hunk 4 above)
Comment (fabianhjr) [reply pkgs/development/python-modules/sip/default.nix]: ah sorry, thought it was a comment on the line addition

File: pkgs/development/python-modules/sip/default.nix (hunk 2 above)
Comment (fabianhjr) [reply pkgs/development/python-modules/sip/default.nix]: Thanks, aligning with python terms on python packages seems like a great move

**Comment by fabianhjr** [comment]
Regarding non-existance of tests can't look into it right now UnU will leave that unresolved

File: pkgs/development/python-modules/sip/default.nix (hunk 4 above)
Comment (fabianhjr) [reply pkgs/development/python-modules/sip/default.nix]: Regarding non-existance of tests can't look into it right now UnU will leave that unresolved

**Pull request merged**